ndonnx/_typed_array/_utils.py

```diff
@@ -57,12 +57,18 @@ def safe_cast(ty: type[T], a: TyArrayBase | bool) -> T:
 
 
 def validate_op_result(
-    x1: TyArrayBase, x2: TyArrayBase, result: T | NotImplementedType, func_name: str
+    x1: TyArrayBase | PyScalar,
+    x2: TyArrayBase | PyScalar,
+    result: T | NotImplementedType,
+    func_name: str,
 ) -> T:
-    """Validate if the provided result is not `NotImplemented` and raise an error if it
-    is."""
+    """Raise an exception if `result` is not `NotImplemented`, otherwise return it."""
+
+    def fmt(x: TyArrayBase | PyScalar) -> str:
+        return str(x) if isinstance(x, PyScalar) else str(x.dtype)
```

Comment:
```suggestion
        return type(x) if isinstance(x, PyScalar) else str(x.dtype)
```

Seems a little strange to produce a dtype for the array and the value for the scalar.